src/dtype.rs

```diff
@@ -99,72 +128,99 @@ pub enum DataType {
 }
 
 impl DataType {
```

Comment:
I wonder if we need to keep `DataType` around at all after this? Couldn't we call the relevant helper functions directly via the (invocations of the) `impl_num_element!` macro and remove this type from our API?

Reply:
This has crossed my mind as well... so what are your concrete suggestions in this case?

(Note that `DataType` has also been removed from error types as a part of migrating `DATA_TYPE` to `IS_POD` which also removes that link)

Reply:
I'd say drop `enum DataType`, pass the `NPY_*` constants or calls to a module-private function `npy_int_type_lookup` directly to the macro.

Reply:
Looks like it's doable. What we'll lose, I guess, is being able to do something like `dtype.datatype() == Some(DataType::Uint32)`. But the question is, is that really useful? (most likely not)